crates/viewer/re_viewer_context/src/drag_and_drop.rs

```diff
@@ -13,166 +51,286 @@ use re_ui::{
 
 use crate::{Contents, Item, ItemCollection};
 
-//TODO(ab): add more type of things we can drag, in particular entity paths
 #[derive(Debug)]
 pub enum DragAndDropPayload {
     /// The dragged content is made only of [`Contents`].
     Contents { contents: Vec<Contents> },
 
+    /// The dragged content is made of entities.
+    Entities { entities: Vec<EntityPath> },
+
     /// The dragged content is made of a collection of [`Item`]s we do know how to handle.
     Invalid,
 }
 
 impl DragAndDropPayload {
     pub fn from_items(selected_items: &ItemCollection) -> Self {
-        if let Ok(contents) = selected_items.try_into() {
+        if let Some(contents) = try_item_collection_to_contents(selected_items) {
             Self::Contents { contents }
+        } else if let Some(entities) = try_item_collection_to_entities(selected_items) {
+            Self::Entities { entities }
```

Comment:
mostly curious (don't mind much), why did you move away from implementing `TryInto` to those methods instead?